weaviate/collection/collection_model.py

```diff
@@ -1,464 +1,91 @@
-
-
-class CollectionObjectModel(CollectionObjectBase, Generic[Model]):
-    def __init__(self, connection: Connection, name: str, model: Type[Model]) -> None:
-        super().__init__(connection, name)
+        self.__consistency_level = consistency_level
         self.__model: Type[Model] = model
-        self._default_props = model.get_non_default_fields(model)
-        self.data = _Data[Model](self)
-        self.query = _GRPCWrapper[Model](self, connection, model)
+        self.__tenant = tenant
 
     @property
     def model(self) -> Type[Model]:
         return self.__model
 
-    def with_tenant(self, tenant: Optional[str] = None) -> "CollectionObjectModel":
-        new_collection = self._with_tenant(tenant)
-        new_collection.data = _Data(new_collection)
-        new_collection.query = _GRPCWrapper(
-            new_collection, new_collection._connection, new_collection.__model
+    def with_tenant(self, tenant: Optional[str] = None) -> "CollectionObjectModel[Model]":
+        return CollectionObjectModel[Model](
+            self._connection, self.name, self.__model, self.config, self.__consistency_level, tenant
         )
-        return new_collection
 
     def with_consistency_level(
         self, consistency_level: Optional[ConsistencyLevel] = None
-    ) -> "CollectionObjectModel":
-        return self._with_consistency_level(consistency_level)
-
-    def _json_to_object(self, obj: Dict[str, Any]) -> _Object[Model]:
-        for ref in self.__model.get_ref_fields(self.__model):
-            if ref not in obj["properties"]:
-                continue
-
-            beacons = obj["properties"][ref]
-            uuids = []
-            for beacon in beacons:
-                uri = beacon["beacon"]
-                assert isinstance(uri, str)
-                uuids.append(uri.split("/")[-1])
-
-            obj["properties"][ref] = uuids
-
-        # weaviate does not save none values, so we need to add them to pass model validation
-        for prop in self._default_props:
-            if prop not in obj["properties"]:
-                obj["properties"][prop] = None
-
-        model_object = _Object[Model](
-            data=self.__model(**obj["properties"]), metadata=_metadata_from_dict(obj)
+    ) -> "CollectionObjectModel[Model]":
+        return CollectionObjectModel[Model](
+            self._connection, self.name, self.__model, self.config, consistency_level, self.__tenant
         )
-        model_object.data.uuid = model_object.metadata.uuid
-        model_object.data.vector = model_object.metadata.vector
-        return model_object
 
 
 class CollectionModel(CollectionBase):
     def __init__(self, connection: Connection):
         super().__init__(connection)
 
-    def create(self, config: CollectionModelConfig[Model]) -> CollectionObjectModel[Model]:
+    def create(
+        self, config: CollectionModelConfig[Model], strict: bool = False
```

Comment:
Why do we need this strict option?

Reply:
ok, but maybe we should use a more descriptive name then?

> Since the client-side checking involves frequent requests to the Weaviate instance for the schema in order to check it appropriately, having this on all the time can lead to a very chatty application where each request is doubled. In a microservice environment where there are many pods all performing this schema validation on each collection for many requests, this could ramp up the number of requests massively.

Does it? I think it only validates that it conforms to the model. I don't see any requests for the schema going out?

Reply:
Would calling it `debug` be more appropriate? Maybe `debug_with_config`? Sadly, I think the size of the changes in the PR have hidden some of the feature additions. I'm going to move the feature additions into a separate PR to make it clearer. Then we can merge this one and discuss on the other 😁

Reply:
that sounds good!